src/TableFunctions/TableFunctionIceberg.cpp

```diff
@@ -1,6 +1,6 @@
 #include "config.h"
 
-#if USE_AWS_S3
```

Comment:
Hi @ucasfl, you forgot about few other places, here is a full patch that I'm using now locally:

```patch
diff --git a/src/Storages/StorageIceberg.cpp b/src/Storages/StorageIceberg.cpp
index 7626f4dbe82..c8bdb5f60dc 100644
--- a/src/Storages/StorageIceberg.cpp
+++ b/src/Storages/StorageIceberg.cpp
@@ -1,5 +1,5 @@
 #include "config.h"
-#if USE_AWS_S3
+#if USE_AWS_S3 && USE_AVRO
 
 #    include <Storages/StorageIceberg.h>
 #    include <Common/logger_useful.h>
diff --git a/src/Storages/StorageIceberg.h b/src/Storages/StorageIceberg.h
index 20afde337c6..166e1c05250 100644
--- a/src/Storages/StorageIceberg.h
+++ b/src/Storages/StorageIceberg.h
@@ -2,7 +2,7 @@
 
 #include "config.h"
 
-#if USE_AWS_S3
+#if USE_AWS_S3 && USE_AVRO
 
 #    include <Storages/IStorageDataLake.h>
 #    include <Storages/S3DataLakeMetadataReadHelper.h>
diff --git a/src/Storages/registerStorages.cpp b/src/Storages/registerStorages.cpp
index 2f3703fb6ea..b2de45ad219 100644
--- a/src/Storages/registerStorages.cpp
+++ b/src/Storages/registerStorages.cpp
@@ -35,8 +35,10 @@ void registerStorageCOS(StorageFactory & factory);
 void registerStorageOSS(StorageFactory & factory);
 void registerStorageHudi(StorageFactory & factory);
 void registerStorageDeltaLake(StorageFactory & factory);
+#if USE_AVRO
 void registerStorageIceberg(StorageFactory & factory);
 #endif
+#endif
 
 #if USE_HDFS
 void registerStorageHDFS(StorageFactory & factory);
@@ -125,8 +127,10 @@ void registerStorages()
     registerStorageOSS(factory);
     registerStorageHudi(factory);
     registerStorageDeltaLake(factory);
+    #if USE_AVRO
     registerStorageIceberg(factory);
-#endif
+    #endif
+    #endif
 
     #if USE_HDFS
     registerStorageHDFS(factory);
diff --git a/src/TableFunctions/TableFunctionIceberg.cpp b/src/TableFunctions/TableFunctionIceberg.cpp
index 16119b8ea78..5600e028527 100644
--- a/src/TableFunctions/TableFunctionIceberg.cpp
+++ b/src/TableFunctions/TableFunctionIceberg.cpp
@@ -1,6 +1,6 @@
 #include "config.h"
 
-#if USE_AWS_S3
+#if USE_AWS_S3 && USE_AVRO
 
 #    include <Storages/StorageIceberg.h>
 #    include <TableFunctions/ITableFunctionDataLake.h>
diff --git a/src/TableFunctions/registerTableFunctions.cpp b/src/TableFunctions/registerTableFunctions.cpp
index 85c0c475bde..1ddaa5ca7a2 100644
--- a/src/TableFunctions/registerTableFunctions.cpp
+++ b/src/TableFunctions/registerTableFunctions.cpp
@@ -29,7 +29,9 @@ void registerTableFunctions()
     registerTableFunctionCOS(factory);
     registerTableFunctionHudi(factory);
     registerTableFunctionDeltaLake(factory);
+#if USE_AVRO
     registerTableFunctionIceberg(factory);
+#endif
     registerTableFunctionOSS(factory);
 
 #endif
```